src/main/resources/javascript/getAllElementsByPath.js

```diff
@@ -60,27 +86,15 @@ function transform(node) {
 		}
 	}
 
-	// these attributes need special treatment
-	extractedAttributes["absolute-x"] = getX(node);
-	extractedAttributes["absolute-y"] = getY(node);
-	extractedAttributes["absolute-width"] = node.getBoundingClientRect().width;
-	extractedAttributes["absolute-height"] = node.getBoundingClientRect().height;
-	if (typeof node.parentNode.getBoundingClientRect === "function") {
-		extractedAttributes["x"] = getX(node) - getX(node.parentNode);
-		extractedAttributes["y"] = getY(node) - getY(node.parentNode);
-		extractedAttributes["width"] = node.getBoundingClientRect().width - node.parentNode.getBoundingClientRect().width;
-		extractedAttributes["height"] = node.getBoundingClientRect().height - node.parentNode.getBoundingClientRect().height;
-	} else {
-		extractedAttributes["x"] = getX(node);
-		extractedAttributes["y"] = getY(node);
-		extractedAttributes["width"] = node.getBoundingClientRect().width;
-		extractedAttributes["height"] = node.getBoundingClientRect().height;
-	}
+	addCoordinates(extractedAttributes, node);
 
 	return extractedAttributes;
 }
 
 function isShown(e) {
+	if (e.nodeType == e.TEXT_NODE) {
```

Comment:
Since we now use this comparison quite a lot (above and below), we can also extract something like:

```javascript
function isText(node) {
    return node.nodeType == node.TEXT_NODE;
}
```

Reply:
Hmm. This would have most value if it is to assume, that this could potentially change in the future (unlikely). But if not, it just introduces complexity, as someone would have to look up, what that method is actually doing. Only a method name like `isNodeTypeTextNode()` would be concise enough ... which I think this is now better...